### DISCUSSION TITLE
libp2p::development_transport got error on Android

### DISCUSSION THREAD

**Discussion by knota-lab** [libp2p/rust-libp2p#2417] [Q&A] [answered]
I try to use libp2p on Android. Execute `libp2p::development_transport` got error `Err(Os { code: 2, kind: NotFound, message: "No such file or directory" })`. Maybe this is a problem about `/etc/resolv.conf` but I'm not sure. How can I fix it?
Thanks.
![image](https://user-images.githubusercontent.com/26835087/147832032-29fb87ec-2ab0-4494-b5e7-537f29575af7.png)

**Comment by dvc94ch** [comment, marked as answer]
We should disable DNS on Android. If you build your own transport without DNS it should work. To actually implement this for Android it would require passing the jnienv and context to all platform specific code and likely some java code as not all of the android apis are jni friendly (for example when you need to create a subclass). the ndk on the other hand is very limited, exposing the bare minimum to create an egl surface for gamedev.

**Reply by knota-lab** [reply]
Thanks a lot. I modified the development_transport as this:
```rust
use libp2p::{identity, core, PeerId, tcp, dns, noise, yamux, mplex, websocket};
use libp2p::Transport;

pub async fn build_transport(
  keypair: identity::Keypair,
) -> std::io::Result<core::transport::Boxed<(PeerId, core::muxing::StreamMuxerBox)>> {
  let transport = {
      let tcp = tcp::TcpConfig::new().nodelay(true);
      // let dns_tcp = dns::DnsConfig::system(tcp).await?;
      // let ws_dns_tcp = websocket::WsConfig::new(dns_tcp.clone());
      // dns_tcp.or_transport(ws_dns_tcp)
      tcp
  };

  let noise_keys = noise::Keypair::<noise::X25519Spec>::new()
      .into_authentic(&keypair)
      .expect("Signing libp2p-noise static DH keypair failed.");

  Ok(transport
      .upgrade(core::upgrade::Version::V1)
      .authenticate(noise::NoiseConfig::xx(noise_keys).into_authenticated())
      .multiplex(core::upgrade::SelectUpgrade::new(
          yamux::YamuxConfig::default(),
          mplex::MplexConfig::default(),
      ))
      .timeout(std::time::Duration::from_secs(20))
      .boxed())
}
```
